src/urllib3/connection.py

```diff
@@ -421,6 +434,43 @@ def connect(self):
             or self.assert_fingerprint is not None
         )
 
+    def set_tls_in_tls_required(self):
+        self.tls_in_tls_required = True
+
+    def _connection_requires_tls_in_tls(self):
+        """
+        Indicates if the current connection requires two TLS connections, one to
+        the proxy and one to the destination server.
+        """
+        return self.tls_in_tls_required
+
+    def _connect_tls_proxy(self, hostname, conn):
+        """
+        Establish a TLS connection to the proxy using the provided SSL context.
+        """
+        proxy_config = self.proxy_config
+        ssl_context = proxy_config.ssl_context
+        if not ssl_context:
+            ssl_context = generate_proxy_ssl_context(
```

Comment:
Should we support configuring `proxy_ca_certs`, `proxy_ssl_version`, etc that live on `ProxyConfig` instead of using `ca_certs` here? Allows for verifying proxy and destination server separately.

Reply:
What I was thinking about is if the user needs to provide custom client certificates or a custom CA to validate the proxy or a custom SSL versoin they should pass the ``proxy_ssl_context`` already prepared.

If not, we'll create one using the default settings we always use. 

That seemed easier and simple. If we introduce ``proxy_ca_certs`` in addition to ``proxy_ssl_context`` I wonder if people will get confused.

Let me know what you think!

Reply:
I think not supporting all options for now is fine. We can always add more later, I suspect it'd be useful to specify client certificate/key instead of having to load it into an SSLContext yourself. Might be something to think about for documentation purposes! :)

I guess I'm then worried about below where all TLS options are used for both the proxy TLS handshake and the destination handshake. Before these values only were meant for the destination handshake.

Maybe some make sense but I think that client certificate doesn't make sense.

Here's my suggestion:
- If we're provided a `proxy_ssl_context` we don't use any values from the `HTTPSConnection`, assume everything is loaded as the user wants.
- If we're not given a `proxy_ssl_context` we can pass in `ca_certs`, `ca_cert_dir`, and `ca_cert_data` but not `keyfile`, `certfile`, and `key_password`. Can document how to achieve this if we're not adding `proxy_certfile`, etc quite yet.